camayoc/tests/qpc/cli/test_sources.py

```diff
@@ -606,6 +622,140 @@ def test_add_with_disable_ssl_negative(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
+        """qpc source add --name {} --cred {} --hosts {} --exclude-hosts {}
+        --type {}"""
+        .format(name, cred_name, hosts, exclude_hosts, source_type)
+    )
+    assert qpc_source_add.expect('Source "{}" was added'.format(name)) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    if hosts.endswith('0/24'):
+        hosts = hosts.replace('0/24', '\[0:255\]')
+    elif hosts.endswith('[1:100]'):
+        hosts = hosts.replace('[1:100]', '\[1:100\]')
+    elif ' ' in hosts:
+        hosts = '",\r\n        "'.join(hosts.split(' '))
+    if exclude_hosts.endswith('0/24'):
+        exclude_hosts = exclude_hosts.replace('0/24', '\[0:255\]')
+    elif exclude_hosts.endswith('0/28'):
+        exclude_hosts = exclude_hosts.replace('0/28', '\[0:15\]')
+    elif exclude_hosts.endswith('[1:100]'):
+        exclude_hosts = exclude_hosts.replace('[1:100]', '\[1:100\]')
+    elif ' ' in exclude_hosts:
+        exclude_hosts = '",\r\n        "'.join(exclude_hosts.split(' '))
+    source_show_and_check(
+        {'name': name},
+        generate_show_output({
+            'cred_name': cred_name,
+            'exclude_hosts': exclude_hosts,
+            'hosts': hosts,
+            'name': name,
+            'port': port,
+            'source_type': source_type,
+        })
+    )
+
+
+@pytest.mark.parametrize('source_type, hosts, exclude_hosts',
+                         VALID_SOURCE_TYPE_HOSTS_WITH_EXCLUDE_HOSTS)
+def test_add_with_cred_hosts_exclude_file(
+        isolated_filesystem, qpc_server_config, hosts, exclude_hosts,
+        source_type):
+    """Add a source with cred and hosts populated on a file.
+
+    :id: 5680125c-c6f6-42b7-bed6-46565ea84902
+    :description: Add a source entry providing the ``--name``, ``--cred``,
+        ``--exclude hosts``, and ``--hosts`` options, the value of the
+        ``--exclude-hosts`` option should be a file.
+    :steps: Run ``qpc source add --name <name> --cred <cred> --hosts
+        <hosts> --exclude-hosts <exclude_hosts_file> --type <type>``
+    :expectedresults: A new source entry is created with the data provided as
+        input.
+    """
+    cred_name = utils.uuid4()
+    name = utils.uuid4()
+    port = default_port_for_source(source_type)
+    cred_add_and_check(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    with open('exclude_hosts_file', 'w') as handler:
+        handler.write(exclude_hosts.replace(' ', '\n') + '\n')
+
+    qpc_source_add = pexpect.spawn(
```

Comment:
Do you think that using source_add_and_check here would be work? https://github.com/quipucords/camayoc/blob/master/camayoc/tests/qpc/cli/utils.py#L185